phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -725,17 +725,17 @@ public List<HRegionLocation> getAllTableRegions(byte[] tableName) throws SQLExce
                 byte[] currentKey = HConstants.EMPTY_START_ROW;
                 do {
                     HRegionLocation regionLocation = ((ClusterConnection)connection).getRegionLocation(
-                            TableName.valueOf(tableName), currentKey, reload);
+                            table, currentKey, false);
                     currentKey = getNextRegionStartKey(regionLocation, currentKey);
                     locations.add(regionLocation);
                 } while (!Bytes.equals(currentKey, HConstants.EMPTY_END_ROW));
                 return locations;
             } catch (org.apache.hadoop.hbase.TableNotFoundException e) {
-                String fullName = Bytes.toString(tableName);
-                throw new TableNotFoundException(fullName);
+                throw new TableNotFoundException(table.getNameAsString());
             } catch (IOException e) {
+                LOGGER.error("Exception encountered in getAllTableRegions for table: "
+                        + table.getNameAsString(), e);
```

Comment:
How about we keep this just above throwing `SQLExceptionInfo.Builder(SQLExceptionCode.GET_TABLE_REGIONS_FAIL)` ? That way we will not log this twice but rather only once when we actually throw the Exception `GET_TABLE_REGIONS_FAIL`

Reply:
Both the exceptions could be different. I think logging both the exceptions should be fine given that this will be in error condition path and not regular code path. @virajjasani  WDYT?

Reply:
Yeah sounds good but from debugging viewpoint, I was thinking maybe in some way if we could differentiate b/ first try and then eventual error after retry? Maybe let's include retry count in log message? I think that would be sufficient to differentiate the first and second attempts (and we can clearly see different root cause for both, if this happens).
Sounds good?